tests/test_prompts.py

```diff
@@ -1,84 +1,80 @@
 
+def test_prompt_model(setup_prompt_object, test_prompt_values):
+    assert setup_prompt_object.is_default is False
+    assert str(setup_prompt_object) == test_prompt_values["label"]
+    assert setup_prompt_object.label == test_prompt_values["label"]
+    assert setup_prompt_object.prompt_value == test_prompt_values["prompt"]
+    assert setup_prompt_object.description == test_prompt_values["description"]
+    assert setup_prompt_object.method == test_prompt_values["method"]
 
-def test_custom_prompt(settings):
-    settings.WAGTAIL_AI_PROMPTS = [
-        {
-            "label": "Custom Prompt",
-            "description": "This is a custom prompt",
-            "prompt": "The prompt",
-            "method": "append",
-        }
-    ]
 
-    assert get_prompts()[0].label == "Custom Prompt"
+def test_get_prompts_returns_default_prompts():
+    prompts = get_prompts()
+    default_prompts_queryset = Prompt.objects.filter(
+        default_prompt_id__isnull=False
+    ).all()
 
+    assert len(default_prompts_queryset) == len(DEFAULT_PROMPTS)
+    assert len(prompts) == len(DEFAULT_PROMPTS)
+    assert prompts[0]["label"] == DEFAULT_PROMPTS[0]["label"]
+    assert prompts[0]["prompt"] == DEFAULT_PROMPTS[0]["prompt"]
 
-def test_extending_prompts(settings):
-    settings.WAGTAIL_AI_PROMPTS = [
-        *DEFAULT_PROMPTS,
-        {
-            "label": "Custom Prompt",
-            "description": "This is a custom prompt",
-            "prompt": "The prompt",
-            "method": "append",
-        },
-    ]
 
-    assert get_prompts()[-1].label == "Custom Prompt"
+def test_editing_default_prompts():
+    prompts = get_prompts()
+    assert prompts[0]["label"] == DEFAULT_PROMPTS[0]["label"]
+    assert prompts[0]["prompt"] == DEFAULT_PROMPTS[0]["prompt"]
 
+    default_prompt = Prompt.objects.get(
+        default_prompt_id=DEFAULT_PROMPTS[0]["default_prompt_id"]
+    )
 
-def test_invalid_prompt(settings):
-    settings.WAGTAIL_AI_PROMPTS = [
-        {
-            "description": "This is a custom prompt",
-            "prompt": "The prompt",
-        }
-    ]
+    default_prompt.prompt = "New Prompt Text"
+    default_prompt.save()
+    assert default_prompt.prompt_value == "New Prompt Text"
 
-    with pytest.raises(ImproperlyConfigured):
-        assert get_prompts()
+    prompts = get_prompts()
+    assert prompts[0]["prompt"] == "New Prompt Text"
+    assert prompts[0]["prompt"] != DEFAULT_PROMPTS[0]["prompt"]
 
 
-def test_invalid_setting_structure(settings):
-    settings.WAGTAIL_AI_PROMPTS = {
-        "label": "Custom Prompt",
-        "description": "This is a custom prompt",
-        "prompt": "The prompt",
-        "method": "append",
-    }
+def test_get_prompts_returns_new_prompts_and_default_prompts(setup_prompt_object):
+    prompts = get_prompts()
+    assert (
+        len(prompts) == len(DEFAULT_PROMPTS) + 1
+    )  # 1 Prompt from setup_prompt_object is added
```

Comment:
Again, this could actually be testing what's bring returned.

```python
from unittest import TestCase

TestCase().assertCountEqual(Prompt.objects.filter(
        default_prompt_id__isnull=False
    ).values_list("label", flat=True),
   [p.label for p in DEFAULT_PROMPTS] + [setup_prompt_object.label]
)
```